debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/sink/SinkConnection.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.sink;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.kafka.common.config.Config;
+import org.apache.kafka.common.config.ConfigValue;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.ReadPreference;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.event.ClusterClosedEvent;
+import com.mongodb.event.ClusterDescriptionChangedEvent;
+import com.mongodb.event.ClusterListener;
+import com.mongodb.event.ClusterOpeningEvent;
+
+import io.debezium.config.Field;
+
+public final class SinkConnection {
```

Comment:
A future jira should be logged to have a single MongoDB connection/client wrapper for both Sink and Source. 

It's ok for now as the connection package in source still needs revisiting and refactoring

Reply:
filed https://issues.redhat.com/browse/DBZ-7995

Reply:
Just a note -- the cluster listener thingy has nothing to do with Atlas. I think I explained why it is used in the other comment.

Reply:
@rk3rn3r Could you please updated the Jira description?